ludwig/data/preprocessing.py

```diff
@@ -1195,6 +1199,12 @@ def build_dataset(
         preprocessing_parameters = feature_name_to_preprocessing_parameters[feature_config[NAME]]
         handle_missing_values(dataset_cols, feature_config, preprocessing_parameters, backend)
 
+    # If we're using LLMs for zero-shot/few-shot learning, update text input features with the prompt
+    # ahead of time so that we can compute metadata and build the preprocessed data correctly.
+    handle_data_augmentation_with_prompt(
```

Comment:
how do we track errors inside this function

Reply:
@skanjila Preprocessing in Ludwig is wrapped around a try/except block, so any errors that happen will trigger the except block and we should be able to diagnose it from there. That being said, it would be good to add some implicit checks where needed